chandra_aca/fov_plot.py

```diff
@@ -0,0 +1,309 @@
+    ax.scatter(gui['yang'], gui['zang'],
+               facecolors='none',
+               edgecolors='green',
+               s=100)
+    for acq_star in acq:
+        acq_box = plt.Rectangle(
+            (acq_star['yang'] - acq_star['halfw'],
+             acq_star['zang'] - acq_star['halfw']),
+            width=acq_star['halfw'] * 2,
+            height=acq_star['halfw'] * 2,
+            color='blue',
+            fill=False)
+        ax.add_patch(acq_box)
+    for mon_box in mon:
+        # starcheck convention was to plot monitor boxes at 2X halfw
+        box = plt.Rectangle(
+            (mon_box['yang'] - (mon_box['halfw'] * 2),
+             mon_box['zang'] - (mon_box['halfw'] * 2)),
+            width=mon_box['halfw'] * 4,
+            height=mon_box['halfw'] * 4,
+            color='orange',
+            fill=False)
+        ax.add_patch(box)
+    ax.scatter(fid['yang'], fid['zang'],
+               facecolors='none',
+               edgecolors='red',
+               linewidth=.5,
+               marker='o',
+               s=175)
+    ax.scatter(fid['yang'], fid['zang'],
+               facecolors='none',
+               edgecolors='red',
+               marker='+',
+               linewidth=.5,
+               s=175)
+
+
+def _plot_field_stars(ax, stars, quat, red_mag_lim=None, bad_stars=None):
+    """
+    Plot plot field stars in yang and zang on the supplied
+    axes object in place.
+
+    :param ax: matplotlib axes
+    :param stars: astropy.table compatible set of records of agasc entries of stars
+    :param quat: attitude quaternion as a Quat
+    :param red_mag_lim: faint limit
+    :param bad_stars: boolean mask of stars to be plotted in red
+    """
+    stars = Table(stars)
+
+    # Add star Y angle and Z angle in arcsec to the stars table
+    yagzags = (radec2yagzag(star['RA_PMCORR'], star['DEC_PMCORR'], quat)
+               for star in stars)
+    yagzags = Table(rows=[(y * 3600, z * 3600) for y, z in yagzags], names=['yang', 'zang'])
+    stars = hstack([stars, yagzags])
+    ok = np.ones(len(stars), dtype='bool')
+
+    if red_mag_lim:
+        nsigma = 3.0
+        mag_error_low_limit = 1.5
+        randerr = 0.26
+        caterr = stars['MAG_ACA_ERR'] / 100.
+        error = nsigma * np.sqrt(randerr*randerr + caterr*caterr)
+        error[error < mag_error_low_limit] = mag_error_low_limit
+        faint = ((stars['MAG_ACA'] >= red_mag_lim)
+                 & ((stars['MAG_ACA'] - error) < red_mag_lim))
+        too_dim_to_plot = ((stars['MAG_ACA'] >= red_mag_lim)
+                            & ((stars['MAG_ACA'] - error) >= red_mag_lim))
+        # All of the indices are on stars, so reindex without too_dim_to_plot
+        stars = stars[~too_dim_to_plot]
+        faint = faint[~too_dim_to_plot]
+        ok = ok[~too_dim_to_plot]
+        ok[faint] = False
+        if bad_stars is not None:
+            bad_stars = bad_stars[~too_dim_to_plot]
+    if bad_stars is not None:
+        ok[bad_stars] = False
+
```

Comment:
That is exactly what I had,  but then discovered that ax.scatter doesn't
seem to take a list for alpha, so I needed multiple scatter calls anyway.
I also didn't know that 'none' world work for primary plot color.  That
might help or maybe I'll just try using patches.
On Nov 16, 2015 6:05 PM, "Tom Aldcroft" <notifications@github.com> wrote:

> In chandra_aca/fov_plot.py
> <https://github.com/sot/chandra_aca/pull/3#discussion_r44998775>:
>
> > +        error = nsigma * np.sqrt(randerr*randerr + caterr*caterr)
> > +        error[error < mag_error_low_limit] = mag_error_low_limit
> > +        faint = ((stars['MAG_ACA'] >= red_mag_lim)
> > +                 & ((stars['MAG_ACA'] - error) < red_mag_lim))
> > +        too_dim_to_plot = ((stars['MAG_ACA'] >= red_mag_lim)
> > +                            & ((stars['MAG_ACA'] - error) >= red_mag_lim))
> > +        # All of the indices are on stars, so reindex without too_dim_to_plot
> > +        stars = stars[~too_dim_to_plot]
> > +        faint = faint[~too_dim_to_plot]
> > +        ok = ok[~too_dim_to_plot]
> > +        ok[faint] = False
> > +        if bad_stars is not None:
> > +            bad_stars = bad_stars[~too_dim_to_plot]
> > +    if bad_stars is not None:
> > +        ok[bad_stars] = False
> > +
>
> The code from 128 to 140 is confusing and I don't have a clue if it is
> correct. I'd suggest starting from scratch by defining a colors array
> that correspond to stars. Then things like:
>
> # Define the color of each star, with black being the default
> colors = np.zeros(len(stars), dtype='S20')
> colors[:] = 'black'
>
> # Check for stars too faint for good catalog but bright enough to be plotted
> ok = (stars['MAG_ACA'] >= red_mag_lim) & (stars['MAG_ACA'] < red_mag_lim + err)
> colors[ok] = FAINT_STAR_COLOR
>
> # Check for stars that are too faint to plot.  The 'none' color stars are not plotted.
> ok = stars['MAG_ACA'] < red_mag_lim + err
> colors[ok] = 'none'
>
> colors[bad_stars] = BAD_STAR_COLOR
> .. etc
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/sot/chandra_aca/pull/3/files#r44998775>.
>